### PR TITLE
Uniform Python and custom mutator API

### PR DESCRIPTION
**Pull request by h1994st**
This pull request provides uniform APIs for both Python module and the custom mutator, as described in #191.

Ready for review

- Add an initialization function for the custom mutator
- Add a pre-save function for Python module
- Add trimming support for the custom mutator by reusing the trimming routines related to Python module
- Add a new structure `struct custom_mutator` to save function pointers of external mutators
- Allow the existence of old `afl_custom_mutator` API in the custom mutator
- Remove `AFL_PYTHON_ONLY` (env) and `python_only` (variable)
- Unify `fuzz` API of the custom mutator and Python module
- Merge the custom mutator into the old python_stage, which is now renamed to custom_mutator_stage
- Merge `python_mutators.md` into `custom_mutator.md`
- Remove `python_mutators.md`
- Merge `examples/python_mutators` into `examples/custom_mutators`
- Remove `examples/python_mutators`
- Update existing examples to demonstrate new APIs

### CODE REVIEW

File: docs/env_variables.md [hunk 1]
```diff
@@ -104,7 +104,7 @@ Then there are a few specific features that are only available in llvm_mode:
     - Setting AFL_LLVM_LAF_SPLIT_COMPARES will split all floating point and
       64, 32 and 16 bit integer CMP instructions
 
-    See llvm_mode/README.laf-intel.md for more information. 
+    See llvm_mode/README.laf-intel.md for more information.
```
Comment (vanhauser-thc) [inline docs/env_variables.md]: ->
See [llvm_mode/README.laf-intel.md](../llvm_mode/README.laf-intel.md) for more information.

Comment (vanhauser-thc) [reply docs/env_variables.md]: that was one we actually overlooked :)

**Comment by vanhauser-thc** [comment]
Looks very good to me.
@andreafioraldi ?

**Comment by andreafioraldi** [comment]
Looks good also for me.
I want to add also `afl_custom_queue_get` that is called at the start of fuzz_one_oroginal() and `afl_custom_queue_new_entry` that is called at the end add_to_queue().
@h1994st if you want to implement also them, you can for me, otherwise I'll do it this weekend.
Regards the last, `afl_custom_havoc_mutation`, I'll do the job myself cause it needs to modify a bit havoc. With this last hook, we can reimplement the stacked smart mutations of AFLSmart as a custom mutator.

**Comment by andreafioraldi** [comment]
@h1994st if you want to add the other two hooks (both in C and python) comment on the issue #191.
I'll merge your work now, ty for the contribution.

**Comment by domenukk** [comment]
If anybody gets bored, feel free to port this to #220 🙃

**Pull request merged**